docs/spec/relayer/ChannelHandlers.tla

```diff
@@ -140,7 +140,64 @@ HandleChanOpenConfirm(chainID, chain, datagrams) ==
     \* otherwise, do not update the chain     
     ELSE chain 
     
+\* Handle "ChanCloseInit" datagrams
+HandleChanCloseInit(chainID, chain, datagrams) ==
+    \* get "ChanCloseInit" datagrams, with a valid channel ID 
+    LET chanCloseInitDgrs == {dgr \in datagrams : 
+                              /\ dgr.type = "ChanCloseInit"
+                              /\ dgr.channelID = GetChannelID(chainID)} IN
+    
+    \* if there are valid "ChanCloseInit" datagrams
+    IF /\ chanCloseInitDgrs /= {} 
+    \* and the channel end is neither UNINIT nor CLOSED
+       /\ chain.connectionEnd.channelEnd.state \notin {"UNINIT", "CLOSED"}
```

Comment:
This is different from ICS04. I guess the reason is that they don't have UNINIT state?Is this the condition used in the Go implementation?

Reply:
Both the ICS004 and the Go implementation checks if the channel end can be found in the store, i.e. if it is non-null. In the TLA+ specification, each channel end is in UNINIT in the initial state of the system (c.f. [Chain.tla](https://github.com/informalsystems/ibc-rs/blob/master/docs/spec/relayer/Chain.tla#L110) and [RelayerDefinitions.tla](https://github.com/informalsystems/ibc-rs/blob/ilina/closing_handshake/docs/spec/relayer/RelayerDefinitions.tla#L160)), hence a channel (also connection) end in state UNINIT denotes a null channel (connection) end.